### PR TITLE
Refgen

### PR DESCRIPTION
**Pull request by eugene-eko2000**

### CODE REVIEW

File: be/referral.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+import (
+	"cleanapp/common"
+	"net/http"
+
+	"github.com/apex/log"
+	"github.com/gin-gonic/gin"
+)
+
+type ReferralQuery struct {
```
Comment (eldarm) [inline be/referral.go:11]: Would this be a good place to documents what these fields are?
Normally this is self-evident, but how one could guess that RefKey = "IP4:Width:Length"?
Or even tougher, that RefValue = ID of the user who referred?

File: be/referral.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	RefKey string `form:"refkey"`
+}
+
+type ReferralResult struct {
+	RefValue string `json:"refvalue"`
```
Comment (eldarm) [inline be/referral.go:16]: And BTW, do you feel that RefValue is such a good name for referring user ID?

File: be/server.go [hunk 3]
```diff
@@ -12,11 +12,13 @@ import (
 const (
 	EndPointUser          = "/update_or_create_user"
 	EndPointReport        = "/report"
-	EndPointReadReport = "/read_report"
+	EndPointReadReport    = "/read_report"
 	EndPointGetMap        = "/get_map"
 	EndPointGetStats      = "/get_stats"
 	EndPointGetTeams      = "/get_teams"
 	EndPointPrivacyAndTOC = "/update_privacy_and_toc"
+	EndPointReadReferral  = "/readreferral"
```
Comment (eldarm) [inline be/server.go:20]: /read_referral ?

File: be/server.go [hunk 4]
```diff
@@ -12,11 +12,13 @@ import (
...
+	EndPointWriteReferral = "/writereferral"
```
Comment (eldarm) [inline be/server.go:21]: In both cases comnvention so far is to use underscor: e.g. /read_referral

File: be/referral.go (hunk 1 above)
Comment (eldarm) [inline be/referral.go:11]: Would this be a good place to documents what these fields are?
Normally this is self-evident, but how one could guess that RefKey = "IP4:Width:Length"?
Or even tougher, that RefValue = ID of the user who referred?

File: be/server.go (hunk 3 above)
Comment (eugene-eko2000) [reply be/server.go:20]: Done.

File: be/server.go (hunk 4 above)
Comment (eugene-eko2000) [reply be/server.go:21]: Done.

File: be/referral.go (hunk 1 above)
Comment (eugene-eko2000) [reply be/referral.go:11]: Good idea, I added the description.

**Comment by eugene-eko2000** [comment]
> Overall looks good. A few questions.

I addresses all your comments. PTAL.

**Review by eldarm — APPROVED** [review]
LGTM

**Pull request merged**